sdks/java/core/src/main/java/org/apache/beam/sdk/testing/WindowFnTestUtils.java

```diff
@@ -67,14 +68,30 @@
   public static <T, W extends BoundedWindow> Map<W, Set<String>> runWindowFn(
       WindowFn<T, W> windowFn,
       List<Long> timestamps) throws Exception {
+    // lift List<Timestamp> into List<TimestampedValue> to factorize implementation
+    List<TimestampedValue<T>> timestampedValues = new ArrayList<>();
+    for (Long timestamp : timestamps){
+      timestampedValues.add(TimestampedValue.of((T) null, new Instant(timestamp)));
```

Comment:
I'm not sure I understand. There is already a `runWindowFn()` version that takes the values, it is `runWindowFnWithValue()`.  Similarly, I did 2 versions of each public method (one that takes only the timestamps and one that takes also the value with a `TimestampedValue`). Indeed, I wanted to keep the existing API and enhance it with a version that also relies on values of elements for window assignment. The `*withValue` version is called with null by the non-value version just to factorize the code.